src/main/fc/core.c

```diff
@@ -674,9 +678,7 @@ bool processRx(timeUs_t currentTimeUs)
     ) {
         if (isUsingSticksForArming()) {
             if (throttleStatus == THROTTLE_LOW) {
-                if (armingConfig()->auto_disarm_delay != 0
-                    && (int32_t)(disarmAt - millis()) < 0
-                ) {
+                if ((armingConfig()->auto_disarm_delay > 0) && (currentTimeUs > disarmAt)) {
```

Comment:
For consistency's sake it probably makes sense to check `autoDisarmDelayUs > 0` here instead of `armingConfig()->auto_disarm_delay > 0`.

Reply:
Good idea - I'll change and push an update.